src/components/Dropdown.js

```diff
@@ -0,0 +1,141 @@
+import { useState, useEffect, useCallback } from "react";
+import SVGIcon from "@/src/components/SVGIcon";
+import {
+  DropdownContainer,
+  DropdownLabel,
+  DropdownButton,
+  DropdownMenu,
+  DropdownItem,
+  ErrorMessage,
+  IconWrapper,
+} from "@/src/components/styles/Dropdown";
+
+export default function Dropdown({
+  label,
+  name,
+  options,
+  buttonText,
+  onOptionClick,
+  required,
+  errorMessage,
+  defaultSelected,
+}) {
+  const [isOpen, setIsOpen] = useState(false);
+  const [selectedOption, setSelectedOption] = useState(defaultSelected || "");
+
+  useEffect(() => {
+    function handleClickOutside(event) {
+      if (isOpen && !event.target.closest(".dropdown-container")) {
+        setIsOpen(false);
+      }
+    }
+
+    document.addEventListener("mousedown", handleClickOutside);
+    return () => {
+      document.removeEventListener("mousedown", handleClickOutside);
+    };
+  }, [isOpen]);
+
+  const toggleDropdown = useCallback((event) => {
+    event.preventDefault();
+    setIsOpen((prev) => !prev);
+  }, []);
+
+  const handleOptionClick = useCallback(
+    (option) => {
+      setSelectedOption(option);
+      setIsOpen(false);
+      onOptionClick(option);
+    },
+    [onOptionClick]
+  );
+
+  const handleKeyDown = useCallback(
+    (event) => {
+      if (
+        ["Enter", " ", "ArrowDown"].includes(event.key) ||
+        (event.key === "Tab" && !isOpen)
+      ) {
+        event.preventDefault();
+        setIsOpen(true);
+        setTimeout(() => {
+          document.querySelector('[role="option"]')?.focus();
+        }, 0);
+      } else if (event.key === "Escape") {
+        setIsOpen(false);
+      }
+    },
+    [isOpen]
+  );
+
+  const handleMenuKeyDown = useCallback(
+    (event) => {
+      const currentIndex = options.findIndex(
+        (option) => option === event.target.innerText
+      );
+      if (event.key === "ArrowDown" || event.key === "ArrowUp") {
+        event.preventDefault();
+        const offset = event.key === "ArrowDown" ? 1 : -1;
+        const nextIndex =
+          (currentIndex + offset + options.length) % options.length;
+        document.getElementById(`option-${nextIndex}`)?.focus();
```

Comment:
You're using the old method to access elements via document quite often. In React this should be avoided.